init.c

```diff
@@ -955,6 +956,13 @@ int luaopen_libcutorch(lua_State *L)
   luaL_setfuncs(L, cutorch_stuff__, 0);
 
   THCState* state = (THCState*)malloc(sizeof(THCState));
+  memset(state, 0, sizeof(THCState));
```

Comment:
THCudaInit now expected the caller to zero-initialize the THCState. This is so that the caller can set the `cudaDeviceAllocator` early (**before** any THCudaMalloc calls).